client/encoding.go

```diff
@@ -19,15 +23,27 @@ type Codec struct {
 	Amino             *codec.LegacyAmino
 }
 
-func MakeCodec(moduleBasics []module.AppModuleBasic) Codec {
+func MakeCodec(moduleBasics []module.AppModuleBasic, extraCodecs []string) Codec {
 	modBasic := module.NewBasicManager(moduleBasics...)
 	encodingConfig := MakeCodecConfig()
 	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
 	std.RegisterInterfaces(encodingConfig.InterfaceRegistry)
 	modBasic.RegisterLegacyAminoCodec(encodingConfig.Amino)
 	modBasic.RegisterInterfaces(encodingConfig.InterfaceRegistry)
-	ethcodec.RegisterInterfaces(encodingConfig.InterfaceRegistry)
-	ethermint.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	for _, c := range extraCodecs {
+		if c == "ethermint" {
```

Comment:
this should be a switch statement

Reply:
do we want any protection here to panic if both ethermint AND injective are attempting to be used 🤔

Reply:
er.. strike that ... the nice thing here is that these are encapsulated in the chains themselves... so unless someone attempts:

```
extra-codecs:
 - ethermint
 - injective
 ```
 
 There should not be an issue.